MyEtherWallet/MyEtherWallet
# Address messages, warnings and gas limits

Closes #900, closes #907, and adds a couple of extras.

### What This Does

* Updates the list of hard-coded messages for addresses from v3 (#907)
    * Moves the message list to config/addressMessages.ts to be somewhat less buried in the code
* Warns users if they're sending to a token contract (#900)
* Uses the hard-coded gas limit provided by addressMessages.ts, if there is one
    * cc @HenryNguyen5, was my implementation on this alright?
* Warns users if they were going to send to themselves, but won't stop them

### How to Test

1. Open the site with the `token-contract-warning` branch
2. Attempt to send to `0x75aa7b0d02532f3833b66c7f0ad35376d373ddf8`, confirm you get a message and your gas limit is set to `300000` with no load time.
3. Attempt to send to `0x0d8775f648430679a709e98d2b0cb6250d2887ef`, confirm you get a message
4. Attempt to send to your own address, confirm you get a message

### Screenshots

<img width="804" alt="screen shot 2018-01-26 at 2 55 54 pm" src="https://user-images.githubusercontent.com/649992/35458272-b3cf9e02-02a9-11e8-8a12-c4559326608e.png">

<img width="802" alt="screen shot 2018-01-26 at 2 56 19 pm" src="https://user-images.githubusercontent.com/649992/35458276-b7ec2064-02a9-11e8-89e1-7c20b179d29f.png">

<img width="797" alt="screen shot 2018-01-26 at 2 57 48 pm" src="https://user-images.githubusercontent.com/649992/35458281-bc5fde1a-02a9-11e8-95c3-f3dccf7fc08a.png">
